### PR TITLE
Implement blank-targetted links health check

### PR DESCRIPTION
**Pull request by alshakero**
https://github.com/Starcounter/DevTools/issues/73

### CODE REVIEW

File: src/views/health-checks.vue [hunk 1]
```diff
@@ -46,6 +66,20 @@ export default {
       }
       return currentWindow;
     },
+    checkBlankTargettedLinks() {
+       const riskyAnchors = [...this.getCurrentWindow().document.querySelectorAll('a[target="_blank"]:not([rel*="noopener"])')];
+       for(const anchor of riskyAnchors) {
+           console.log(anchor);
```
Comment (warpech) [inline src/views/health-checks.vue]: Is this on purpose?

File: src/views/health-checks.vue [hunk 2]
```diff
@@ -3,9 +3,9 @@
         <div v-if="compatible">
             <h1>Health checks</h1>
```
Comment (warpech) [inline src/views/health-checks.vue]: In Firefox, this popup is not scrollable after displaying the list of health checks in BlendingEditor:

image

File: src/views/health-checks.vue [hunk 3]
```diff
@@ -3,9 +3,9 @@
...
             <p>This tab is a beta feature. Currently it only checks if all the Custom Elements you're using are imported. Soon we'll add more checks you can use for your apps conveniently.
-            <p><button v-on:click="checkElementsHealth">Run checks</button></p>
-            <hr />
-            <table>
+            <h3>1. Undefined Custom Elements: checks whether all the CEs you use are imported correctly</h3>
+            <p><button v-on:click="checkElementsHealth">Run check</button></p>
+            <table v-if="customElementsHealth.length > 0">
```
Comment (warpech) [inline src/views/health-checks.vue]: Please add some message if `customElementsHealth.length == 0`

File: src/views/health-checks.vue [hunk 4]
```diff
@@ -15,6 +15,25 @@
...
+                    <th>DOM Path</th>
+                    <th>innerText</th>
+                </tr>
+                <tr v-for="anchor in riskyAnchors" v-bind:key="anchor.name">
+                    <td>{{anchor.href}}</td>
+                    <td>{{anchor.path}}</td>
+                    <td>{{anchor.innerText}}</td>
+                </tr>
+            </table>
+            <p v-if="riskyAnchors.length > 0"><small>Results are logged. You can hover the results in Chrome DevTools console to highlight the element</small></p>
```
Comment (warpech) [inline src/views/health-checks.vue]: Please add some message if `riskyAnchors.length == 0`

File: src/views/health-checks.vue [hunk 5]
```diff
@@ -15,6 +15,25 @@
...
+            
+            <h3>2. Common security check: blank-targetted links with no <code>rel="noopener"</code> (see <a href="https://developers.google.com/web/tools/lighthouse/audits/noopener" target="_blank" rel="noopner"> here</a>)</h3>
+            <p><button v-on:click="checkBlankTargettedLinks">Run check</button></p>
+            <table v-if="riskyAnchors.length > 0">
+                <tr>
+                    <th>HREF</th>
+                    <th>DOM Path</th>
+                    <th>innerText</th>
+                </tr>
+                <tr v-for="anchor in riskyAnchors" v-bind:key="anchor.name">
```
Comment (warpech) [inline src/views/health-checks.vue]: Can you display all `a[target="_blank"]` and make use of CSS classes `ok-bg` and `error-bg` to highlight the ones without `noopener`?

**Review by warpech — CHANGES_REQUESTED** [review]

File: src/views/health-checks.vue (hunk 1 above)
Comment (alshakero) [reply src/views/health-checks.vue]: Yes so you could hover over then in DevTools and see them highlighted like this:
image

**Comment by alshakero** [comment]
Closed as outdated.

**Pull request closed without merging**